File: domain/src/testFixtures/java/com/study/platform/support/fake/FakeFailedPostSyncRepository.java
package com.study.platform.support.fake;

import com.study.platform.domain.post.model.FailedPostSync;
import com.study.platform.domain.post.model.FailedPostSyncRepository;

import java.util.ArrayList;
import java.util.List;

public class FakeFailedPostSyncRepository implements FailedPostSyncRepository {

    private final List<FailedPostSync> store = new ArrayList<>();

    @Override
    public FailedPostSync save(FailedPostSync failedPostSync) {
        store.add(failedPostSync);
        return failedPostSync;
    }

    public List<FailedPostSync> getSaved() {
        return store;
    }
}

File: domain/src/testFixtures/kotlin/com/study/platform/support/fake/FakeFailedPostSyncRepository.kt
package com.study.platform.support.fake

import com.study.platform.domain.post.model.FailedPostSync
import com.study.platform.domain.post.model.FailedPostSyncRepository

class FakeFailedPostSyncRepository : FailedPostSyncRepository {

    private val store: MutableList<FailedPostSync> = ArrayList()

    override fun save(failedPostSync: FailedPostSync): FailedPostSync {
        store.add(failedPostSync)
        return failedPostSync
    }

    override fun saveIndependently(failedPostSync: FailedPostSync): FailedPostSync = save(failedPostSync)

    fun getSaved(): List<FailedPostSync> = store
}
